src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -438,6 +488,8 @@ private static bool CheckConstraintsSingleType(TypeSymbol type, CheckConstraints
             Location location,
             DiagnosticBag diagnostics)
         {
+            type = (NamedTypeSymbol)type.TupleUnderlyingTypeOrSelf();
```

Comment:
@alrz I've learned not to bet against review feedback :-)
Let's add the suggested test.

---
In reply to: [92724285](https://github.com/dotnet/roslyn/pull/15641#discussion_r92724285) [](ancestors = 92724285)